var/spack/repos/builtin/packages/python/package.py

```diff
@@ -37,6 +37,50 @@ def make_pyvenv_cfg(python_pkg: Package, venv_prefix: str) -> str:
     return "\n".join(lines) + "\n"
 
 
+def find_python_in_prefix(spec: "spack.spec.Spec", prefix: Optional[str] = None) -> Executable:
```

Comment:
Since it's a free-standing function, we should probably error if `spec.name` is not in an expected set of names.[^1]

[^1]: not sure if we want to include `python-venv` or not

Reply:
Why do we need the check here?  What if they want to find an embedded python in some package we don't know about?  What's the harm?

Reply:
Hmm, if this is meant to find e.g. a `python3` within an arbitrary package, then I don't get:
```
suffixes = [spec.version.up_to(2), spec.version.up_to(1), ""]
```
cause that assumes that arbitrary package follows the same versioning as Python?

Reply:
I also wonder if the special case on `platform-python` wouldn't be best left outside of a generic function.

Reply:
I made it so that this will error the spec name is not `python`.  This is really just a refactor to pull the logic out ... I don't think at this point it needs to be more generic.  platform-python is a `python` that we support so I don't really know why we need to pull it out.